database/src/test/java/database/service/SqlDatabaseServiceTest.java

```diff
@@ -0,0 +1,274 @@
+    private final String UNABLE_CREATE_TABLE =
+            "Unable to create table. Please check if it already exists: Table 'student' already exists";
+    private final String TABLE_NOT_EXIST = "Table does not exist: Student";
+    private MockedStatic<DriverManager> driverManagerMock;
+
+    @BeforeEach
+    void setUp() throws SQLException {
+        firstStudent = new Student("FirstName1 LastName1", 5.0);
+        secondStudent = new Student("FirstName2 LastName2", 4.5);
+        thirdStudent = new Student("FirstName3 LastName3", 5.0);
+        fourthStudent = new Student("FirstName1 LastName1", 5.0);
+        sqlDatabaseService = new SqlDatabaseService(DATABASE_URL, DB_USER_NAME, DB_PASSWORD, DATABASE_NAME);
+    }
+
+    @AfterEach
+    void tearDown() {
+        try {
+            sqlDatabaseService.deleteTable(Student.class);
+        } catch (Exception e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    @Test
+    void createTableTest() throws SQLException {
+        Class<? extends BaseEntity> entityClass = Student.class;
+
+        assertTrue(sqlDatabaseService.createTable(entityClass));
+        assertTrue(sqlDatabaseService.checkTableExists(DATABASE_NAME, "Student"));
+    }
+
+    @Test
+    void createTableFailedTest() {
+        Class<? extends BaseEntity> entityClass = Student.class;
+        sqlDatabaseService.createTable(entityClass);
+
+        CreationDatabaseException exception = assertThrows(CreationDatabaseException.class, () ->
+                sqlDatabaseService.createTable(entityClass));
+        assertEquals(UNABLE_CREATE_TABLE, exception.getMessage());
+    }
+
+    @Test
+    void deleteTableTest() throws SQLException {
+        Class<? extends BaseEntity> entityClass = Student.class;
+        sqlDatabaseService.createTable(entityClass);
+
+        assertTrue(sqlDatabaseService.deleteTable(entityClass));
+        assertFalse(sqlDatabaseService.checkTableExists(DATABASE_NAME, "Student"));
+    }
+
+    @Test
+    void addNewRecordToTablePositiveTest() throws SQLException {
+        Class<? extends BaseEntity> entityClass = Student.class;
+
+        sqlDatabaseService.createTable(entityClass);
+
+        Student receivedFirstStudent = sqlDatabaseService.addNewRecordToTable(firstStudent);
+        assertEquals(firstStudent, receivedFirstStudent);
+        assertEquals(1, receivedFirstStudent.getId());
+
+        Student receivedSecondStudent = sqlDatabaseService.addNewRecordToTable(secondStudent);
+        assertEquals(secondStudent, receivedSecondStudent);
+        assertEquals(2, receivedSecondStudent.getId());
+    }
+
+    @Test
+    void addNewRecordWhenDatabaseDoesNotExistTest() {
+        DatabaseDoesNotExistException exception = assertThrows(DatabaseDoesNotExistException.class, () ->
+                sqlDatabaseService.addNewRecordToTable(firstStudent));
+
+        assertEquals(TABLE_NOT_EXIST, exception.getMessage());
```

Comment:
Please use `TABLE_NOT_EXIST` constant from service class and check with assertTrue(exception.getMessage().startswith(TABLE_NOT_EXIST)).